metaflow/plugins/argo/argo_events_decorator.py

```diff
@@ -0,0 +1,173 @@
+from metaflow.decorators import FlowDecorator
+from metaflow.exception import MetaflowException
+from metaflow.util import is_stringish
+
+# TODO: Support dynamic parameter mapping through a context object that exposes
+#       flow name and user name similar to parameter context
+
+
+# TODO: This decorator interface can be a top-level decorator to support similar
+#       implementations for Netflix Maestro, AWS Step Functions, Apache Airflow...
+class ArgoEventsDecorator(FlowDecorator):
+    name = "trigger"
+    defaults = {
+        "event": None,
+        "events": [],
+        "options": {},  # TODO: introduce support for options
+    }
+
+    def flow_init(
+        self, flow, graph, environment, flow_datastore, metadata, logger, echo, options
+    ):
+        self.triggers = []
+        if sum(map(bool, (self.attributes["event"], self.attributes["events"]))) > 1:
+            raise MetaflowException(
+                "Specify only one of *event* or *events* "
+                "attributes in *@trigger* decorator."
+            )
+        elif self.attributes["event"]:
+            # event attribute supports the following formats -
+            #     1. event='table.prod_db.members'
+            #     2. event={'name': 'table.prod_db.members',
+            #               'parameters': {'alpha': 'member_weight'}}
+            if is_stringish(self.attributes["event"]):
+                self.triggers.append({"name": str(self.attributes["event"])})
+            elif isinstance(self.attributes["event"], dict):
+                if "name" not in dict(self.attributes["event"]):
+                    raise MetaflowException(
+                        "The *event* attribute for *@trigger* is missing the "
+                        "*name* key."
+                    )
+                self.triggers.append(self.attributes["event"])
+            else:
+                raise MetaflowException(
+                    "Incorrect format for *event* attribute in *@trigger* decorator. "
+                    "Supported formats are string and dictionary - \n"
+                    "@trigger(event='foo') or @trigger(event={'name': 'foo', "
+                    "'parameters': {'alpha': 'beta'}})"
+                )
+        elif self.attributes["events"]:
+            # events attribute supports the following formats -
+            #     1. events='table.prod_db.members AND table.prod_db.metadata'
```

Comment:
I think we have previously talked about not wanting to do some complex syntax like this? I actually liked what I thought we had talked about last time:
  - things within `events` are "AND"
  - you can have multiple decorators to do "OR"

Reply:
that is still the case - things within `events` are `AND`. This is just an explicit flavor of that syntax. Multiple decorators is not supported at this point for this release. Can easily add it as a follow-up when the need presents itself.